[stable/envoy] Fix deployment spec

hypnoglow opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:

- Remove serviceName as it is not defined for Deployment

- Move terminationGracePeriodSeconds from Deployment spec to pod
template spec

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

unguiculus commented on stable/envoy/templates/deployment.yaml:
```diff
@@ -44,6 +40,7 @@ spec:
 {{ toYaml .Values.securityContext | indent 8 }}
 {{- if .Values.priorityClassName }}
       priorityClassName: "{{ .Values.priorityClassName }}"
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
```
Should be outside the if statement.

hypnoglow replied:
woops.. fixed.

hypnoglow committed:
[stable/envoy] Fix deployment spec

- Remove serviceName as it is not defined for Deployment

- Move terminationGracePeriodSeconds from Deployment spec to pod
template spec

Signed-off-by: Igor Zibarev <zibarev.i@gmail.com>
